crypto/src/merkle.rs

```diff
@@ -99,36 +107,41 @@ impl<U> FromIterator<HashOf<U>> for MerkleTree<U> {
 }
 
 impl<T> MerkleTree<T> {
-    /// Constructs new instance of the merkle tree
+    /// Construct [`MerkleTree`].
     pub const fn new() -> Self {
         MerkleTree {
             root_node: Node::Empty,
         }
     }
 
-    /// Returns leaf node
+    /// Get the hash of the `idx`-th leaf node.
     pub fn get_leaf(&self, idx: usize) -> Option<HashOf<T>> {
-        self.root_node.get_leaf_inner(idx).ok()
+        self.iter().nth(idx)
```

Comment:
this search can, most probably, be optimized to `log(n)`

Reply:
I would leave a comment at least

Reply:
Addressed in 6c93c3c6992dac9e4c880f2a87d83c9b1e089f6d